File: Project Euler/p686.py
import math

lo, hi = math.log10(1.23), math.log10(1.24)
const = math.log10(2)

n = 678910
j = 0
cnt = 0
while cnt != n:
    j += 1
    if lo < j * const - math.floor(j * const) < hi:
        cnt += 1

print(j)

File: Project Euler/p686.cpp
#include <cmath>
#include <cstdio>
using namespace std;

int main() {
    double lo = log10(1.23), hi = log10(1.24), mult = log10(2);

    int n = 678910, j = 0, cnt = 0;
    while (cnt != n) {
        j++;
        double cur = j * mult - floor(j * mult);
        if (lo < cur && cur < hi) cnt++;
    }
    printf("%d\n", j);
}